ndcube/utils/wcs.py

```diff
@@ -429,3 +429,36 @@ def array_indices_for_world_objects(wcs, axes=None):
         array_index = convert_between_array_and_pixel_axes(pixel_index, wcs.pixel_n_dim)
         array_indices[oinds] = tuple(array_index[::-1])  # Invert from pixel order to array order
     return tuple(ai for ai in array_indices if ai)
+
+
+def validate_wcs(source_wcs, target_wcs):
```

Comment:
I would name this something more descriptive like `compare_wcs_physical_types`

Reply:
I came to this conversation late and I know @adwaitbhope has already made this change.  But I think the function name should say what is wants to do, rather than how it does it.  If not `validate_wcs`, then something like `assert_wcs_compatible`.

In the future we may want to do this comparison in a more sophisticated way, e.g. taking into account invariant axes, or comparing against the axes object types.  If the function name does not refer to its implementation but rrather the role it performs, changing how we validate should have minimal impact on the code outside this function.

Reply:
I suspect that any further questions we are going to want to ask of the WCSes, i.e. invariant axes etc are going to want to be different functions. Also by convention functions which start `assert_` tend to raise assertion errors if the comparison fails.

Maybe the best thing to do here is change the first line of the docstring to make the scope of this function clear?

Reply:
Sounds fine to me.